#!/usr/bin/python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "usv_gazebo_plugins"


gen = ParameterGenerator()

mapping_enum = gen.enum([gen.const("Linear", int_t, 0, "Linear thrust mapping"),
                         gen.const("GLF", int_t, 1, "General logistic function thrust mapping")],
                        "Mapping type")
gen.add("bow_thruster_fwd", double_t, 0,
        "Bow thruster max forward force", 1.0, 0, 10.0)
gen.add("bow_thruster_rev", double_t, 0,
        "Bow thruster max reverse force", 1.0, 0, 10.0)
gen.add("propeller_fwd", double_t, 0,
        "Propeller max forward force", 1.0, 0, 10.0)
gen.add("propeller_rev", double_t, 0,
        "Propeller max reverse force", 1.0, 0, 10.0)
gen.add("sway_factor", double_t, 0,
        "Propeller max reverse force", 0, 0, 1.0)
gen.add("mapping_type", int_t, 0, "Mapping type for the thrusters", 0, 0, 1, edit_method=mapping_enum)

exit(gen.generate(PACKAGE, "usv_gazebo_nhawigatora_thrust", "Thrust"))
